airflow/serialization/helpers.py

```diff
@@ -38,7 +39,19 @@ def is_jsonable(x):
         else:
             return True
 
+    max_size = conf.getint("core", "max_templated_field_size")
     if not is_jsonable(template_field):
-        return str(template_field)
+        serialized = str(template_field)
+        if len(serialized) > max_size:
+            return (
+                "Value removed due to size. "
```

Comment:
Should we truncate vs remove it completely?

Reply:
My concern was the information about how to configure the size. Would it make sense to have a truncated data alongside the info about how to increase the config value?

Reply:
Seems reasonable to me, as a prefix so it's visible.